typescript/agentkit/src/action-providers/compound/utils.ts

```diff
@@ -0,0 +1,515 @@
+export const getTokenDecimals = async (
+  wallet: EvmWalletProvider,
+  tokenAddress: `0x${string}`,
+): Promise<number> => {
+  if (!tokenAddress) {
+    throw new Error("Token address is undefined");
+  }
+
+  const decimals = await wallet.readContract({
+    address: tokenAddress,
+    abi: ERC20_ABI,
+    functionName: "decimals",
+  });
+
+  return Number(decimals);
+};
+
+/**
+ * Get token symbol from contract
+ *
+ * @param wallet - The wallet provider instance
+ * @param tokenAddress - The address of the token contract
+ * @returns The symbol of the token
+ */
+export const getTokenSymbol = async (
+  wallet: EvmWalletProvider,
+  tokenAddress: `0x${string}`,
+): Promise<string> => {
+  if (!tokenAddress) {
+    throw new Error("Token address is undefined");
+  }
+
+  const symbol = await wallet.readContract({
+    address: tokenAddress,
+    abi: ERC20_ABI,
+    functionName: "symbol",
+  });
+
+  return symbol as string;
+};
+
+/**
+ * Get token balance for an address
+ *
+ * @param wallet - The wallet provider instance
+ * @param tokenAddress - The address of the token contract
+ * @returns The token balance as a bigint
+ */
+export const getTokenBalance = async (
+  wallet: EvmWalletProvider,
+  tokenAddress: `0x${string}`,
+): Promise<bigint> => {
+  if (!tokenAddress) {
+    throw new Error("Token address is undefined");
+  }
+
+  const balance = await wallet.readContract({
+    address: tokenAddress,
+    abi: ERC20_ABI,
+    functionName: "balanceOf",
+    args: [wallet.getAddress()],
+  });
+
+  return balance as bigint;
+};
+
+/**
+ * Get collateral balance for an address
+ *
+ * @param wallet - The wallet provider instance
+ * @param cometAddress - The address of the Comet contract
+ * @param tokenAddress - The address of the token contract
+ * @returns The collateral balance as a bigint
+ */
+export const getCollateralBalance = async (
+  wallet: EvmWalletProvider,
+  cometAddress: `0x${string}`,
+  tokenAddress: `0x${string}`,
+): Promise<bigint> => {
+  if (!tokenAddress || !cometAddress) {
+    throw new Error("Address undefined");
+  }
+
+  const balance = await wallet.readContract({
+    address: cometAddress,
+    abi: COMET_ABI,
+    functionName: "collateralBalanceOf",
+    args: [await wallet.getAddress(), tokenAddress],
+  });
+
+  return balance as bigint;
+};
+
+/**
+ * Format amount from decimals to human readable
+ *
+ * @param amount - The amount to format
+ * @param decimals - The number of decimals
+ * @returns The formatted amount as a string
+ */
+export const formatAmountFromDecimals = (amount: bigint, decimals: number): string => {
+  return formatUnits(amount, decimals);
+};
+
+/**
+ * Format amount with decimals from human readable
+ *
+ * @param amount - The amount to format
+ * @param decimals - The number of decimals
+ * @returns The formatted amount as a bigint
+ */
+export const formatAmountWithDecimals = (amount: string, decimals: number): bigint => {
+  return parseUnits(amount, decimals);
+};
```

Comment:
Why use these functions over the viem function they are wrapping? In other words – why not use `formatUnits` and `parseUnits` directly?